challenge_receptionist/src/challenge_receptionist/find_empty_seat.py

```diff
@@ -37,6 +50,75 @@ def __repr__(self):
         return "SeatsInRoomDesignator({}, {}, {}, {})".format(self.robot, self.seat_ids, self.room, self.name)
 
 
+class SeatVolumeNamesDesignator(ds.Designator):
+    def __init__(self, robot, seat_entity_designator, name=None, debug=False):
+        super(SeatVolumeNamesDesignator, self).__init__(resolve_type=[str], name=name)
+
+        self.robot = robot
+
+        ds.check_type(seat_entity_designator, Entity)
+        self.seat_entity_designator = seat_entity_designator
+
+        self.debug = debug
+
+    def _resolve(self):
+        if self.debug:
+            import ipdb;ipdb.set_trace()
+
+        try:
+            seat_entity = self.seat_entity_designator.resolve()  # type: Entity
+            seat_volume_names = sorted([k for k in seat_entity.volumes.keys() if k.endswith('seat')])
+
+            return seat_volume_names
+        except AttributeError:
+            return None
```

Comment:
Can be made shorter

if seat_entity:
    seat_volume_na.....

Reply:
Better safe than sorry